test/test_ssl.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016: Alignak team, see AUTHORS.txt file for contributors
+#
+# This file is part of Alignak.
+#
+# Alignak is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Alignak is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Alignak.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This file test the SSL on daemons
+"""
+
+import subprocess
+from time import sleep
+import requests
+import shutil
+from alignak_test import AlignakTest
+
+
+class TestSsl(AlignakTest):
+    """
+    This class test the SSL on daemons
+    """
+    def _get_subproc_data(self, name):
+        try:
+            print("Try to end %s" % name)
+            self.procs[name].send_signal(2)
+            self.procs[name].send_signal(15)
+            self.procs[name].wait()
+        except Exception as err:
+            print("prob on terminate and wait subproc %s: %s" % (name, err))
+        data = {}
+        data['out'] = self.procs[name].stdout.read()
+        data['err'] = self.procs[name].stderr.read()
+        data['rc'] = self.procs[name].returncode
+        return data
+
+    def setUp(self):
+        # openssl genrsa -passout pass:wazabi -out certificate_test.key 2048
+        # openssl req -new -x509 -days 3650 -key certificate_test.key -out certificate_test.csr
+        # openssl dhparam -out dhparams.pem 2048
+        shutil.copy('cfg/ssl/certificate_test.csr', '/tmp/')
+        shutil.copy('cfg/ssl/certificate_test.key', '/tmp/')
+        shutil.copy('cfg/ssl/dhparams.pem', '/tmp/')
+        self.procs = {}
+        self.ssl_installed = True
+        try:
+            from OpenSSL import SSL
+        except ImportError:
+            self.ssl_installed = False
+            print "Install pyopenssl"
+            subprocess.call("pip install pyopenssl", shell=True)
```

Comment:
The goal is to have other tests works without pyopenssl not installed and run this test with pyopenssl. Like this we test with and without (it's a good idea because yesterday without pyopenssl alignak has broken, but I have fixed it)